root/constants.js

```diff
@@ -7,6 +7,24 @@
  * later version: http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
+export const ACCESS_SCOPE_PROFILE: 1 = 1;
+export const ACCESS_SCOPE_EMAIL: 2 = 2;
+export const ACCESS_SCOPE_TAG: 4 = 4;
+export const ACCESS_SCOPE_RATING: 8 = 8;
+export const ACCESS_SCOPE_COLLECTION: 16 = 16;
+export const ACCESS_SCOPE_SUBMIT_ISRC: 64 = 64;
+export const ACCESS_SCOPE_SUBMIT_BARCODE: 128 = 128;
```

Comment:
(Not stupid at all.) If you do `const x = 1` Flow will see the type of x as `number`, whereas in `const x: 1 = 1` it'll see it as the number literal `1`.

Here's an example where it could conceivably make a difference: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjOA7AzgFzAEEBhEgUQGVKB9SkgeQAVyaAlIgFQEkA5AcQBcYABxgAvKIDcmHAWJkqtes1aMAMuvIkeDXsICMANglhjM1PgCeABwCmCitTqMWlCajBgAPmGv24KEclF1V2Lj5+Tx8-WztA4OcVFhoNLR1uPWiLLDxCKGxTAAoAQ1xhUidlVyoASgkAPjAAV2wAEzsoAEtsOzaLAqKDACYAZlqZIA (if you remove the `: 8` or `: 16` bits, the type of `ACCESS_SCOPES` is just `number` and `fn` accepts anything.)

Reply:
I still don’t get why Flow needs annotation here, can it not deduce that `x` will be `1` forever?

Reply:
Not currently. There are some open issues for it, e.g. https://github.com/facebook/flow/issues/2639 and https://github.com/facebook/flow/issues/3439.